pkg/daemon/tproxy_linux.go

```diff
@@ -0,0 +1,375 @@
+			klog.Fatalf("Unrecoverable error while accepting connection: %s", err)
+			return
+		}
+		go handleRedirectFlow(conn)
+	}
+}
+
+func (c *Controller) StartTProxyTCPPortProbe() {
+
+	probePorts := map[string]interface{}{}
+
+	pods, err := c.getTProxyConditionPod(false)
+	if err != nil {
+		return
+	}
+
+	for _, pod := range pods {
+		iface := ovs.PodNameToPortName(pod.Name, pod.Namespace, util.OvnProvider)
+		nsName, err := ovs.GetInterfacePodNs(iface)
+		if err != nil || nsName == "" {
+			continue
+		}
+
+		for _, podIP := range pod.Status.PodIPs {
+			customVPCPodIPToNs.Store(podIP.IP, nsName)
+			for _, container := range pod.Spec.Containers {
+				if container.ReadinessProbe != nil {
+					if tcpSocket := container.ReadinessProbe.TCPSocket; tcpSocket != nil {
+						if port := tcpSocket.Port.String(); port != "" {
+							probePorts[port] = nil
+						}
+					}
+				}
+
+				if container.LivenessProbe != nil {
+					if tcpSocket := container.LivenessProbe.TCPSocket; tcpSocket != nil {
+						if port := tcpSocket.Port.String(); port != "" {
+							probePorts[port] = nil
+						}
+					}
+				}
+			}
+
+			for port := range probePorts {
+				probePortInNs(podIP.IP, port, true, nil)
+			}
+		}
+	}
+}
+
+func (c *Controller) runTProxyConfigWorker() {
+	protocols := getProtocols(c.protocol)
+	for _, protocol := range protocols {
+		c.reconcileTProxyRoutes(protocol)
+	}
+}
+
+func (c *Controller) reconcileTProxyRoutes(protocol string) {
+	family := getFamily(protocol)
+	if err := addRuleIfNotExist(family, TProxyOutputMark, TProxyOutputMask, util.TProxyRouteTable); err != nil {
+		return
+	}
+
+	if err := addRuleIfNotExist(family, TProxyPreroutingMark, TProxyPreroutingMask, util.TProxyRouteTable); err != nil {
+		return
+	}
+
+	dst := GetDefaultRouteDst(protocol)
+	if err := addRouteIfNotExist(family, util.TProxyRouteTable, &dst); err != nil {
+		return
+	}
+}
+
+func (c *Controller) cleanTProxyConfig() {
+	protocols := getProtocols(c.protocol)
+	for _, protocol := range protocols {
+		c.cleanTProxyRoutes(protocol)
+		c.cleanTProxyIPTableRules(protocol)
+	}
+}
+
+func (c *Controller) cleanTProxyRoutes(protocol string) {
+	family := getFamily(protocol)
+	if err := deleteRuleIfExists(family, TProxyOutputMark); err != nil {
+		klog.Errorf("delete tproxy route rule mark %v failed err: %v ", TProxyOutputMark, err)
+	}
+
+	if err := deleteRuleIfExists(family, TProxyPreroutingMark); err != nil {
+		klog.Errorf("delete tproxy route rule mark %v failed err: %v ", TProxyPreroutingMark, err)
+	}
+
+	dst := GetDefaultRouteDst(protocol)
+	if err := delRouteIfExist(family, util.TProxyRouteTable, &dst); err != nil {
+		klog.Errorf("delete tproxy route rule mark %v failed err: %v ", TProxyPreroutingMark, err)
+	}
+}
+
+func getFamily(protocol string) int {
+	family := netlink.FAMILY_ALL
+	if protocol == kubeovnv1.ProtocolIPv4 {
+		family = netlink.FAMILY_V4
+	} else if protocol == kubeovnv1.ProtocolIPv6 {
+		family = netlink.FAMILY_V6
+	}
+	return family
+}
```

Comment:
Can we use `util.ProtocolToFamily()`?

Reply:
1

Reply:
We should investigate whether a linter is capable of performing this task